src/mscorlib/src/System/Array.cs

```diff
@@ -1559,31 +1559,23 @@ public static void Reverse<T>(T[] array)
         {
             if (array == null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
-            Reverse(array, 0, array.Length);
+
+            // The Span ctor might perform an unnecessary type check if T is a reference type,
```

Comment:
This type check would not be just unnecessary. It would actually be a correctness problem for co-variant arrays.